src/rust/runtime/mod.rs

```diff
@@ -148,6 +154,7 @@ impl SharedDemiRuntime {
             ephemeral_ports: EphemeralPorts::default(),
             timer: SharedTimer::new(now),
             network_table: NetworkQueueTable::default(),
+            pending_ops: HashMap::<QDesc, HashMap<TaskHandle, YielderHandle>>::new(),
```

Comment:
Something to thinkg about... We could set an initial capacity for this structure, so as to prevent initial performance degradation while the hashmap grows.